src/crush/CrushTreeDumper.h

```diff
@@ -167,6 +167,14 @@ namespace CrushTreeDumper {
 
     bool is_touched(int id) const { return touched.count(id) > 0; }
 
+    void set_root(const string& bucket) {
+      roots.clear();
+      if (crush->name_exists(bucket)) {
+	int i = crush->get_item_id(bucket);
+	roots.insert(i);
+      }
+    }
+
```

Comment:
roots is empty if the bucket doesn't exist?

Reply:
yes, nothing to show